bin/pre-commit

```diff
@@ -34,8 +37,8 @@ BRANCH_NAME=$(git branch | grep '*' | sed 's/* //')
 if [ $BRANCH_NAME != '(no branch)' ]; then
     cd $ROOT
 
-    echo "updating autogen files..."
-    bin/regenerate_files.py
+    #echo "updating autogen files..."
+    #bin/regenerate_files.py
```

Comment:
as above, we need something to do proper generation somewhere. i don't have strong informed opinions on where, however.

Reply:
I don't know if people are more likely to install the pre-submit hook (I forget sometimes), or to 
run make / test. 

But it's clear make will need to have the target - so maybe we can start with having it in the makefile.